### ISSUE TITLE
prefer-await-to-then doesn't enforce .then/.catch/.finally

### ISSUE THREAD

**Issue by bisubus** [eslint-community/eslint-plugin-promise#294]
### Description

Currently the rule is not enforced if a chain in async/await is correctly established. It could possibly have `strict` option to get rid of promise methods and prefer `await` completely.

### Steps to Reproduce

```js
async function doQuery() {
  await query
    .then(() => { console.log('complete') })
    .catch(() => { console.log('error') })
}
```

**Expected behavior:**

Both `then` and `catch` trigger linter error.

**Actual behavior:**

The error is triggered with `then` and only if `await` is missing.

### Versions

[Please fill this in if you are submitting a bug report]

- Node version: v16.13.2
- ESLint version: 7.32.0
- eslint-plugin-promise version: 6.0.0

### Additional Information

Related: https://github.com/xjamundx/eslint-plugin-promise/issues/187

**Comment by thodoo** [comment]
Wanting to enforce this was the reason I found this plugin. I also expected the same behavior.

**Comment by netotz** [comment]
> Wanting to enforce this was the reason I found this plugin

same lmao
